### ISSUE TITLE
Tables with similar data are unwantedly linked.

### ISSUE THREAD

**Issue by ExTheSea** [swimlane/ngx-datatable#759]
**I'm submitting a ...**
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**
When you supply two instances of a component containing a ngx-datatable with similar data it will act the same. If the data is identical sorting one table will also sort the other table. If only a few rows are identical it will cause visual bugs where if sorting table A it puts the identical rows from table B in the same position as in table A. This often overlays other rows which are in these positions.

**Expected behavior**
No matter how similar the data is, sorting one instance of a table should not sort the other one.

**Reproduction of the problem**
I could only reproduce the bigger case which happens when all items are identical in plnkr:
http://plnkr.co/edit/M6lrwbVRbp0w1Dga0zLz?p=info
The other case only happens in my more complex example where e.g. every array has a type.

Basically:
* Create a component with a ngx-datatable and the rows as input.
* Supply the component with similar content
* Sort one table and watch the other one

**What is the motivation / use case for changing the behavior?**
I have a use case where there are two tables on a page which display the same type of data, just a subset of it. This is impossible to do with ngx-datatable because any action on one table will affect the other one.

**Please tell us about your environment:**
Windows, IntelliJ, npm

* **Table version:** 9.1.0

* **Angular version:** 4.1.2

* **Browser:** Tested on Chrome 58.0.3029.110 and Firefox 53.0.2 though probably affects all Browsers

* **Language:** Typescript 2.3.2 with es2016 setup

**Comment by amcdnl** [comment]
You need to pass new arrays, we don't recreate the objects, we simply re-arrange the original one.

**Comment by ExTheSea** [comment]
@amcdnl 
While this explains my simple plnkr it doesn't explain what's happening in my project. It also doesn't explain what's happening in this modified plnkr:
http://plnkr.co/edit/UgnPS8hIVsEbqstiJIxY?p=preview
Here i am creating a subset of the first array using .filter which by definition creates a new Array. Still, sorting the second table will also sort the first one. This looks like maybe the rows are bound to the object instances inside the array which seems odd.

**Comment by ExTheSea** [comment]
A sort of workaround i just figured out btw would be to deep copy your filtered array like
JSON.parse(JSON.stringify(yourArray.filter(....)))
So it seems like ngx-datatable doesn't really sort on the array instance but rather the object instances on some kind of table sort array shared between the tables?
